lib/identity/client.js

```diff
@@ -141,6 +139,35 @@ class Client extends PartialClient {
     const resultQuery = await this.storage.search(request)
     return resultQuery
   }
+  async updateSecret(secret) {
+    // Search for all records of the same type
+    const request = new Search(true)
+    let searchType = `tozny.secret.${SECRET_UUID}.${secret.secretType}.${secret.secretName}`
+    request.match({ type: searchType })
+    const resultQuery = await this.storage.search(request)
+    const secretList = await resultQuery.next()
+    // Find the Record Tagged Latest
+    let recordId
+    for (var i = 0; i < secretList.length; i++) {
+      if (secretList[i].meta.plain['version'] == 'latest') {
+        recordId = secretList[i].meta.recordId
+      }
+    }
+    // Update latest Record to be new version number
+    const versionNumber = resultQuery.totalResults
+    const updatedMetaVersion = {
+      secretType: secret.secretType,
+      secretName: secret.secretName,
+      description: secret.description,
+      version: `v${versionNumber}`,
+    }
+    const read = await this.storage.readRecord(recordId)
+    read.meta.plain = updatedMetaVersion
+    await this.storage.updateRecord(read)
```

Comment:
I don't think we have a need to or should be updating records. When "updating secrets" It is really just creating a new record of the same type and record name.